### ISSUE TITLE
The username should be used for Merge Request

### ISSUE THREAD

**Issue by Shad0w1nk** [gitlabhq/gitlabhq#9914]
I really think the username should be use when accepting the merge request as opposed to using the name: https://github.com/gitlabhq/gitlabhq/blob/master/app/models/repository.rb#L500

Gitlab is already converting the user.name from commit to the User.name in the UI based on the User.username. It would only be fair to use User.username during the commit of the merge.

**Comment by stanhu** [comment]
I'm confused why that would be the correct thing to do. As far as I can tell, the name and and e-mail address are passed to rugged/libgit2 so that the proper `Author` looks like:

```
Name <e-mail address>
```

Just as `git config` intended with `user.name` and `user.email`.

**Comment by Shad0w1nk** [comment]
I was coming from the angle where gitlab does the mapping in the ui from the git user.name to gitlab name through the gitlab username.

When committing through git we use the gitlab username as the git user.name and when someone accept a merge request it get committed using the gitlab name instead of the gitlab username. It also make the git log a bit inconsistent as the workflow through git will be using gitlab username and the one through gitlab will be using gitlab name.

We can change the gitlab name to be the username in gitlab but that feel more like an hack since gitlab already do the mapping in the ui.

What do you think? Or is there a different workaround?

**Comment by stanhu** [comment]
I'm confused: GitLab uses the e-mail address to determine the username, not the git `user.name`. You should use a real name in `user.name` (e.g. John Smith), not a GitLab username.

**Comment by Shad0w1nk** [comment]
That's alright. We use the username for user.name in the git commit but when you accept a merge request through Gitlab, the full name is used in the commit instead of the username. I saw that Gitlab was mapping the user.name from git to the Gitlab full name using the Gitlab username so it would make sense to use the username. I see this is a different requirement per organization.

We moved away from Gitlab so I don't have any need for such a fix/feature. Thanks for the quick response.
